Cannot find Custom Resource Definition error

I've run the MySQL sample and get the following output:

```shell
java -jar target/mysql-schema-sample-1.6.2-SNAPSHOT.jar
08:50:46.066 INFO  MySQLSchemaOperator.java      (22) MySQL Schema Operator starting
08:50:47.668 INFO  Operator.java                 (112) Registered Controller: 'SchemaController' for CRD: 'class io.javaoperatorsdk.operator.sample.Schema' for namespaces: [all/client namespace]
```

I've copied the sample code into a standalone Gradle project and built a shadow JAR, which gives me the following error (I've got a slightly different log format):

```shell
java -jar libs/demo-operator-0.0.1-SNAPSHOT.jar
Exception in thread "main" java.lang.IllegalArgumentException: No custom resource has been found for controller io.javaoperatorsdk.operator.sample.SchemaController
	at io.javaoperatorsdk.operator.config.runtime.RuntimeControllerMetadata.getCustomResourceClass(RuntimeControllerMetadata.java:32)
	at io.javaoperatorsdk.operator.config.runtime.AnnotationConfiguration.getCustomResourceClass(AnnotationConfiguration.java:49)
	at io.javaoperatorsdk.operator.Operator.registerController(Operator.java:78)
	at io.javaoperatorsdk.operator.Operator.registerControllerForAllNamespaces(Operator.java:56)
	at io.javaoperatorsdk.operator.sample.MySQLSchemaOperator.main(MySQLSchemaOperator.java:27)
```

Has anyone else come across this and what's the solution? Am I just missing something bleedingly obvious from the docs/samples?

Ah ha, that solves it, although the generated code causes warnings regarding unchecked exceptions:

```shell
❯ ./gradlew installDist

> Task :compileJava
warning: Supported source version 'RELEASE_8' from annotation processor 'org.gradle.api.internal.tasks.compile.processing.TimeTrackingProcessor' less than -source '11'
/Users/ashley/code/strataform/strataform-operator/build/generated/sources/annotationProcessor/java/main/io/javaoperatorsdk/operator/sample/SchemaDoneable.java:8: warning: [unchecked] unchecked method invocation: constructor <init> in class CustomResourceDoneable is applied to given types
    super(resource,function);
         ^
  required: T,Function<T,T>
  found:    Schema,Function
  where T is a type-variable:
    T extends CustomResource declared in class CustomResourceDoneable
/Users/ashley/code/strataform/strataform-operator/build/generated/sources/annotationProcessor/java/main/io/javaoperatorsdk/operator/sample/SchemaDoneable.java:8: warning: [unchecked] unchecked conversion
    super(resource,function);
                   ^
  required: Function<T,T>
  found:    Function
  where T is a type-variable:
    T extends CustomResource declared in class CustomResourceDoneable
3 warnings

BUILD SUCCESSFUL in 3s
5 actionable tasks: 4 executed, 1 up-to-date
```

Is this something we can easily fix?

--- a/operator-framework/src/main/java/io/javaoperatorsdk/operator/config/runtime/ControllerAnnotationProcessor.java
+++ b/operator-framework/src/main/java/io/javaoperatorsdk/operator/config/runtime/ControllerAnnotationProcessor.java
@@ -4,6 +4,7 @@ import static io.javaoperatorsdk.operator.config.runtime.RuntimeControllerMetada
 import static io.javaoperatorsdk.operator.config.runtime.RuntimeControllerMetadata.DONEABLES_RESOURCE_PATH;
 
 import com.google.auto.service.AutoService;
+import com.squareup.javapoet.AnnotationSpec;
 import com.squareup.javapoet.ClassName;
 import com.squareup.javapoet.JavaFile;
 import com.squareup.javapoet.MethodSpec;
@@ -149,6 +150,10 @@ public class ControllerAnnotationProcessor extends AbstractProcessor {
                 .addParameter(customResourceType, "resource")
                 .addParameter(Function.class, "function")
                 .addStatement("super(resource,function)")
+                .addAnnotation(
+                    AnnotationSpec.builder(SuppressWarnings.class)
+                        .addMember("value", "$S", "unchecked")
+                        .build())
                 .build();
 
         final TypeSpec typeSpec =

--- a/operator-framework/src/main/java/io/javaoperatorsdk/operator/config/runtime/ControllerAnnotationProcessor.java
+++ b/operator-framework/src/main/java/io/javaoperatorsdk/operator/config/runtime/ControllerAnnotationProcessor.java
@@ -107,12 +107,8 @@ public class ControllerAnnotationProcessor extends AbstractProcessor {
         controllersResourceWriter.add(
             controllerClassSymbol.getQualifiedName().toString(),
             CustomResource.class.getCanonicalName());
-        return;
-      }
-
-      if (resourceType == null) {
         System.out.println(
-            "No defined a resource type for '"
+            "No defined resource type for '"
                 + controllerClassSymbol.getQualifiedName()
                 + "': ignoring!");
         return;

--- a/operator-framework/src/main/java/io/javaoperatorsdk/operator/config/runtime/ControllerAnnotationProcessor.java
+++ b/operator-framework/src/main/java/io/javaoperatorsdk/operator/config/runtime/ControllerAnnotationProcessor.java
@@ -35,7 +35,7 @@ import javax.tools.Diagnostic;
 import javax.tools.JavaFileObject;
 
 @SupportedAnnotationTypes("io.javaoperatorsdk.operator.api.Controller")
-@SupportedSourceVersion(SourceVersion.RELEASE_8)
+@SupportedSourceVersion(SourceVersion.RELEASE_11)
 @AutoService(Processor.class)
 public class ControllerAnnotationProcessor extends AbstractProcessor {
 

--- a/operator-framework/src/test/resources/compile-fixtures/ControllerImplemented2InterfacesExpected.java
+++ b/operator-framework/src/test/resources/compile-fixtures/ControllerImplemented2InterfacesExpected.java
@@ -4,6 +4,7 @@ import io.fabric8.kubernetes.api.builder.Function;
 import io.fabric8.kubernetes.client.CustomResourceDoneable;
 
 public class MyCustomResourceDoneable extends CustomResourceDoneable<ControllerImplemented2Interfaces.MyCustomResource> {
+    @SuppressWarnings("unchecked")
     public MyCustomResourceDoneable(ControllerImplemented2Interfaces.MyCustomResource resource, Function function) {
         super(resource, function);
     }

--- a/operator-framework/src/test/resources/compile-fixtures/ControllerImplemented2InterfacesExpected.java
+++ b/operator-framework/src/test/resources/compile-fixtures/ControllerImplemented2InterfacesExpected.java
@@ -2,6 +2,7 @@ package io;
 
 import io.fabric8.kubernetes.api.builder.Function;
 import io.fabric8.kubernetes.client.CustomResourceDoneable;
+import java.lang.SuppressWarnings;
 
 public class MyCustomResourceDoneable extends CustomResourceDoneable<ControllerImplemented2Interfaces.MyCustomResource> {
     @SuppressWarnings("unchecked")

--- a/operator-framework/src/test/resources/compile-fixtures/ControllerImplementedIntermediateAbstractClassExpected.java
+++ b/operator-framework/src/test/resources/compile-fixtures/ControllerImplementedIntermediateAbstractClassExpected.java
@@ -4,6 +4,7 @@ import io.fabric8.kubernetes.api.builder.Function;
 import io.fabric8.kubernetes.client.CustomResourceDoneable;
 
 public class MyCustomResourceDoneable extends CustomResourceDoneable<AbstractController.MyCustomResource> {
+    @SuppressWarnings("unchecked")
     public MyCustomResourceDoneable(AbstractController.MyCustomResource resource, Function function) {
         super(resource, function);
     }

--- a/operator-framework/src/test/resources/compile-fixtures/ControllerImplementedIntermediateAbstractClassExpected.java
+++ b/operator-framework/src/test/resources/compile-fixtures/ControllerImplementedIntermediateAbstractClassExpected.java
@@ -2,6 +2,7 @@ package io;
 
 import io.fabric8.kubernetes.api.builder.Function;
 import io.fabric8.kubernetes.client.CustomResourceDoneable;
+import java.lang.SuppressWarnings;
 
 public class MyCustomResourceDoneable extends CustomResourceDoneable<AbstractController.MyCustomResource> {
     @SuppressWarnings("unchecked")

--- a/operator-framework/src/test/resources/compile-fixtures/MultilevelControllerExpected.java
+++ b/operator-framework/src/test/resources/compile-fixtures/MultilevelControllerExpected.java
@@ -4,6 +4,7 @@ import io.fabric8.kubernetes.api.builder.Function;
 import io.fabric8.kubernetes.client.CustomResourceDoneable;
 
 public class MyCustomResourceDoneable extends CustomResourceDoneable<MultilevelController.MyCustomResource> {
+    @SuppressWarnings("unchecked")
     public MyCustomResourceDoneable(MultilevelController.MyCustomResource resource, Function function) {
         super(resource, function);
     }

--- a/operator-framework/src/test/resources/compile-fixtures/MultilevelControllerExpected.java
+++ b/operator-framework/src/test/resources/compile-fixtures/MultilevelControllerExpected.java
@@ -2,6 +2,7 @@ package io;
 
 import io.fabric8.kubernetes.api.builder.Function;
 import io.fabric8.kubernetes.client.CustomResourceDoneable;
+import java.lang.SuppressWarnings;
 
 public class MyCustomResourceDoneable extends CustomResourceDoneable<MultilevelController.MyCustomResource> {
     @SuppressWarnings("unchecked")

